src/roblox_api/instance.rs

```diff
@@ -91,6 +91,35 @@ impl LuaInstance {
         Ok(child)
     }
 
+    fn get_descendants(&self) -> rlua::Result<Vec<LuaInstance>> {
+        let tree = self.tree.lock().unwrap();
+
+        let instance = tree.get_by_ref(self.id).ok_or_else(|| {
+            rlua::Error::external("Cannot call GetDescendants() on a destroyed instance")
+        })?;
+
+        let mut descendants = Vec::new();
+        let mut stack = Vec::new();
+        let mut current = Some(instance);
+
+        while let Some(current_instance) = current {
```

Comment:
Keeping a stack and `current` is weird. We should jut use `while let Some(current_instance) = stack.pop()` in our loop and drop the `current` variable. We can initialize the stack with our top-level instance!